src/react-contenteditable.js

```diff
@@ -20,13 +20,25 @@ export default class ContentEditable extends React.Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    return !this.htmlEl || nextProps.html !== this.htmlEl.innerHTML ||
-            this.props.disabled !== nextProps.disabled;
+    // We need not rerender if the change of props simply reflects the user's
+    // edits. Rerendering in this case would make the cursor/caret jump.
+    return (
+      // Rerender if there is no element yet... (somehow?)
+      !this.htmlEl
+      // ...or if html really changed... (programmatically, not by user edit)
+      || ( nextProps.html !== this.htmlEl.innerHTML
+        && nextProps.html !== this.props.html )
+      // ...or if editing is enabled or disabled.
+      || this.props.disabled !== nextProps.disabled
+    );
   }
 
   componentDidUpdate() {
     if ( this.htmlEl && this.props.html !== this.htmlEl.innerHTML ) {
-     this.htmlEl.innerHTML = this.props.html;
+      // Perhaps React's VDOM (which gets outdated because we often prevent
+      // rerendering) thought that updating was not required. So we update the
+      // element manually now.
```

Comment:
I would like to merge this PR. But it seems to me that this comment is wrong, isn't it ?
If react thought that updating was not needed, `componentDidUpdate()` wouldn't be called anyway. It did update the component, but did not change the DOM, because its VDOM was out of date. Is that it, @slorber ?

So the comment should be:
```js
// Perhaps React (whose VDOM gets outdated because we often prevent
// rerendering) did not update the DOM. So we update it manually now.
```

Tell me if I'm wrong.